MoreLinq/Batch.cs

```diff
@@ -146,8 +179,9 @@ IEnumerable<TResult> Batch(int size)
                     // Return the last bucket with all remaining elements
                     if (bucket != null && count > 0)
                     {
-                        Array.Resize(ref bucket, count);
-                        yield return resultSelector(bucket);
+                        var newArray = bucketFactory(count);
+                        Array.Copy(bucket, 0, newArray, 0, count);
```

Comment:
does the same as `Array.Resize(ref bucket, count)` but manually to use `bucketFactory`